packages/SwingSet/test/stores/test-durabilityChecks.js

```diff
@@ -0,0 +1,216 @@
+import { test } from '../../tools/prepare-test-env-ava.js';
+
+// eslint-disable-next-line import/order
+import { Far } from '@agoric/marshal';
+import { makeFakeVirtualStuff } from '../../tools/fakeVirtualSupport.js';
+
+const { vom, cm } = makeFakeVirtualStuff({ cacheSize: 3 });
+
+const { makeScalarBigMapStore, makeScalarBigSetStore } = cm;
+
+const { makeKind, makeDurableKind } = vom;
+
+function makeHolderInstance(state) {
+  return {
+    init(value = null) {
+      state.held = value;
+    },
+    self: Far('holder', {
+      hold(value) {
+        state.held = value;
+      },
+    }),
+  };
+}
+
+const virtualHolderMaker = makeKind(makeHolderInstance);
+const durableHolderMaker = makeDurableKind(makeHolderInstance);
+
+const aString = 'zorch!';
+const aVirtualObject = virtualHolderMaker();
+const aDurableObject = durableHolderMaker();
+const aRemotableObject = Far('what', {
+  aMethod() {
+    return 'remote whatever';
+  },
+});
+const aVirtualStore = makeScalarBigMapStore('vstore');
+const aDurableStore = makeScalarBigMapStore('dstore', { durable: true });
+
+const anObjectFullOfVirtualStuff = harden({
+  aString,
+  aVirtualObject,
+  aDurableObject,
+  aRemotableObject,
+  aVirtualStore,
+  aDurableStore,
+});
+const anObjectFullOfDurableStuff = harden({
+  aString,
+  aDurableObject,
+  aDurableStore,
+});
+const anArrayFullOfVirtualStuff = harden([
+  aString,
+  aVirtualObject,
+  aDurableObject,
+  aRemotableObject,
+  aVirtualStore,
+  aDurableStore,
+]);
+const anArrayFullOfDurableStuff = harden([
+  aString,
+  aDurableObject,
+  aDurableStore,
+]);
+
+function m(s) {
+  return { message: s };
+}
+
+// prettier-ignore
+test('durability checks', t => {
+  const failKey = f => t.throws(f, m('key is not durable'));
+  const failVal = f => t.throws(f, m('value is not durable'));
+  const failHold = f => t.throws(f, m('value for "held" is not durable'));
+  const passKey = f => t.notThrows(f);
+  const passVal = f => t.notThrows(f);
+  const passHold = f => t.notThrows(f);
+
+  const virtualMap = makeScalarBigMapStore('vmap');
+  const durableMap = makeScalarBigMapStore('dmap', { durable: true });
+
+  passKey(() => virtualMap.init(aString, 'simple string key'));
```

Comment:
Since doing a `delete` on a durable collection with a non-durable key fails just like any other attempt to delete a key that is not in the collection (and since a non-durable can't have ever been put in there the delete *will* fail), there's no special key check for such a case nor any real reason for one.  Consequently there's nothing special to test for either.